0002/README.md

```diff
@@ -0,0 +1,209 @@
+solution is likely necessary for several reasons: to achieve hermeticity for
+SLSA 4, to generate a quality SBOM, to provide a better UX, and to make it
+possible to rebuild on other platforms.
+
+Work includes:
+
+*   Proof-of-concept SLSA extensions to the Sigstore-based solutions above to
+    reach SLSA 4 for specific languages (go, Python, etc.) These will hopefully
+    prove ideas and provide more data, but we do not expect to support these
+    long term.
+*   Proposal for how to generalize and scale ecosystem-specific solutions to
+    achieve SLSA 4 and generate high quality SBOMs that can be maintained by
+    each community independently, without risk of accidentally breaking SLSA
+    guarantees.
+
+#### Milestone: Provenance distribution model is decided, with at least one representative implementation (Python and/or NPM)
+
+Consumers have a standard mechanism for finding and retrieving provenance of the
+packages they use within each packaging ecosystem. The tentative plan is to
+distribute provenance in-band next to the artifact, plus have an out-of-band
+mechanism for prototyping and supporting attestations from parties other than
+the producer. This work also includes aligning with SBOM, since the requirements
+are very similar. With this milestone, motivated consumers can act upon the
+provenance themselves.
+
+Issue: [#269](https://github.com/slsa-framework/slsa/issues/269)
+
+#### Milestone: SLSA policy model is decided, to detect or prevent publishing of packages containing unauthorized changes, with at least one representative implementation (Python and/or NPM)
+
+SLSA policies detect or prevent unauthorized changes for packages within the
+ecosystem. When opted-in, package maintainers must upload provenance alongside
+the package proving that the package was built directly from the canonical
+source repository. At first this may just display results on a dashboard,
+similar to [pyreadiness.org](https://pyreadiness.org/3.10/), but will hopefully
+be integrated directly into PyPI/NPM eventually. With this milestone, software
+producers can opt-in to automatically protect themselves against insider risk,
+and software consumers can gain assurance for opted-in packages, but the package
+registry is still trusted.
+
+Sub-problems:
+
+-   Policy model, which securely maps package name to canonical source
+    repository. For example, [requests](https://pypi.org/project/requests/) maps
+    to https://github.com/psf/requests. This metadata already exists for some
+    packages, but (a) it is not structured and (b) package maintainers can
+    mutate it at will. The solution must protect against malicious changes while
+    silently allowing official changes in source repository. One idea is to have
+    a Trust On First Use (TOFU) model with a mechanism for source repo
+    migrations.
+
+-   Implementation for each specific ecosystem, starting with Python/PyPI. This
```

Comment:
we mention Python/PyPI explicitly here but talk about PyPI/Python or npm above -- is Python the chosen representative implementation? Should we remove references to npm?

Reply:
Removed NPM. (We might also propose something for NPM if time permits, but no need to list that here.)

We are working with @di, who is a PSF board member. He's on board with the idea in general, without committing anything. That said, we still have to discussed it with the Python community and go through the PEP process. My thinking was that we want to start with a design and prototype implementation to give us something concrete to discuss.

Reply:
Rephrasing looks good, thanks. I think we (SLSA) should be open about what level of buy-in we have (lest folks assume this is a done-deal). But that might fit as part of the detailed planning i.e., listing Python outreach and PEP authoring as tasks?

Reply:
Good point. Added.